Framework/Geometry/inc/MantidGeometry/IObjComponent.h

```diff
@@ -84,7 +84,7 @@ class MANTID_GEOMETRY_DLL IObjComponent : public virtual IComponent {
   virtual void initDraw() const = 0;
 
   /// Returns the shape of the Object
-  virtual const std::shared_ptr<const IObject> shape() const = 0;
+  virtual const std::shared_ptr<const IObject> &shape() const = 0;
```

Comment:
Similarly, did `cppcheck` warn on this? I think for return types and `shared_ptr` we are better off returning by value as we don't know how the implementing object is creating the `shared_ptr` to return and returning a reference forces it to store it.